### ISSUE TITLE
auto_remove supported but not allowed

### ISSUE THREAD

**Issue by lindycoder** [docker/docker-py#1579]
**VERSION**
docker==2.2.1
docker-pycreds==0.2.1
Python 2.7.11
Client:
 Version:      17.05.0-ce-rc1
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   2878a85
 Built:        Tue Apr 11 20:55:05 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.05.0-ce-rc1
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   2878a85
 Built:        Tue Apr 11 20:55:05 2017
 OS/Arch:      linux/amd64
 Experimental: true

**OS**
OS X El Capitan - 10.11.6

**ISSUE**
I tried to use client.containers.run(...., detach=True, auto_remove=True) but i get the following:
```
Error
Traceback (most recent call last):
  File ".../python2.7/site-packages/nose/suite.py", line 208, in run
    self.setUp()
  File ".../python2.7/site-packages/nose/suite.py", line 291, in setUp
    self.setupContext(ancestor)
  File ".../python2.7/site-packages/nose/suite.py", line 314, in setupContext
    try_run(context, names)
  File ".../python2.7/site-packages/nose/util.py", line 478, in try_run
    return func()
  File ".../__init__.py", line 15, in setup
    extra_hosts=hosts)
  File ".../python2.7/site-packages/docker/models/containers.py", line 660, in run
    detach=detach, **kwargs)
  File ".../python2.7/site-packages/docker/models/containers.py", line 703, in create
    create_kwargs = _create_container_args(kwargs)
  File ".../python2.7/site-packages/docker/models/containers.py", line 884, in _create_container_args
    raise create_unexpected_kwargs_error('run', kwargs)
TypeError: run() got an unexpected keyword argument 'auto_remove'
```

I dug up a bit and found that it breaks here:
https://github.com/docker/docker-py/blob/master/docker/models/containers.py#L899
```
    # All kwargs should have been consumed by this point, so raise
    # error if any are left
    if kwargs:
        raise create_unexpected_kwargs_error('run', kwargs) 
```

The arg should have been picked up in this loop:
https://github.com/docker/docker-py/blob/master/docker/models/containers.py#L879
```
    for key in copy.copy(kwargs):
        if key in RUN_HOST_CONFIG_KWARGS:
            host_config_kwargs[key] = kwargs.pop(key)
```

But the list here does include auto_remove 
https://github.com/docker/docker-py/blob/master/docker/models/containers.py#L816

While the HostConfig class clearly has it in its kwargs:
https://github.com/docker/docker-py/blob/master/docker/types/containers.py#L103

**OBSERVATION**
It would seem that there's a discrepancy between the kwargs list constant and the actual kwarg list

**EXTRA NOTE**
I know run() has a remove= arg but it says it can't be used with detach=True so i'm expecting auto_remove to work

**Comment by shin-** [comment]
see #1504 and related issues.

**Comment by lindycoder** [comment]
right, will be fixed by #1545 
should i close this?
